src/Native/Runtime/gcrhenv.cpp

```diff
@@ -818,6 +818,13 @@ void GCToEEInterface::RestartEE(bool /*bFinishedGC*/)
 {
     FireEtwGCRestartEEBegin_V1(GetClrInstanceId());
 
+#if defined(TARGET_ARM) || defined(TARGET_ARM64)
+    // Flush the store buffers on all CPUs, to ensure that they all see changes made
+    // by the GC threads. This only matters on weak memory ordered processors as 
+    // the strong memory ordered processors wouldn't have reordered the relevant reads.
```

Comment:
Could you please add additional clarification from https://github.com/dotnet/runtime/pull/42281/ too?